app/src/main/java/org/oppia/app/player/state/hintsandsolution/HintsAndSolutionExplorationManagerFragment.kt

```diff
@@ -0,0 +1,64 @@
+package org.oppia.app.player.state.hintsandsolution
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import org.oppia.app.fragment.InjectableFragment
+import javax.inject.Inject
+
+private const val KEY_EXPLORATION_ID = "EXPLORATION_ID"
+private const val KEY_NEW_AVAILABLE_HINT_INDEX = "NEW_AVAILABLE_HINT_INDEX"
+private const val KEY_ALL_HINTS_EXHAUSTED = "ALL_HINTS_EXHAUSTED"
+
+/**
+ * ManagerFragment of [ExplorationFragment] that observes data provider that retrieve Exploration State.
+ */
+class HintsAndSolutionExplorationManagerFragment : InjectableFragment() {
+  @Inject
+  lateinit var hintsAndSolutionExplorationManagerFragmentPresenter: HintsAndSolutionExplorationManagerFragmentPresenter
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    fragmentComponent.inject(this)
+  }
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    val args =
+      checkNotNull(arguments) { "Expected arguments to be passed to HintsAndSolutionFragment" }
+    val id =
+      checkNotNull(args.getString(KEY_EXPLORATION_ID)) { "Expected id to be passed to HintsAndSolutionFragment" }
+    val newAvailableHintIndex =
+      checkNotNull(args.getInt(KEY_NEW_AVAILABLE_HINT_INDEX)) { "Expected hint index to be passed to HintsAndSolutionFragment" }
+    val allHintsExhausted =
+      checkNotNull(args.getBoolean(KEY_ALL_HINTS_EXHAUSTED)) { "Expected if hints exhausted to be passed to HintsAndSolutionFragment" }
+
+    return hintsAndSolutionExplorationManagerFragmentPresenter.handleCreateView(
+      id,
+      newAvailableHintIndex,
+      allHintsExhausted
+    )
+  }
+
+  companion object {
+
+    fun newInstance(
+      explorationId: String,
+      newAvailableHintIndex: Int,
+      allHintsExhausted: Boolean
+    ): HintsAndSolutionExplorationManagerFragment {
+      val explorationManagerFragment = HintsAndSolutionExplorationManagerFragment()
+      val args = Bundle()
+      args.putString(KEY_EXPLORATION_ID, explorationId)
```

Comment:
Please update these as per meeting suggestions...(here  and elsewhere) PTAL on https://docs.google.com/document/d/17mjInnMroI17jlSkn0GhDo-iVw9heF1z8FtjY_saUl0/edit

Reply:
Done.